ui/apps/platform/src/Containers/ComplianceEnhanced/Dashboard/Widgets/ComplianceByCluster.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { useState } from 'react';
+import { Chart, ChartAxis, ChartBar, ChartGroup, ChartLabelProps } from '@patternfly/react-charts';
+
+import { LinkableChartLabel } from 'Components/PatternFly/Charts/LinkableChartLabel';
+import useResizeObserver from 'hooks/useResizeObserver';
+import {
+    defaultChartHeight,
+    defaultChartBarWidth,
+    patternflySeverityTheme,
+} from 'utils/chartUtils';
+
+import {
+    LOW_SEVERITY_COLOR,
+    MODERATE_MEDIUM_SEVERITY_COLOR,
+    IMPORTANT_HIGH_SEVERITY_COLOR,
+    CRITICAL_SEVERITY_COLOR,
+} from 'constants/visuals/colors';
+
+import WidgetCard from '../../../Dashboard/Widgets/WidgetCard';
+
+const labelLinkCallback = ({ datum }: ChartLabelProps, data: ComplianceData) => {
+    return typeof datum === 'number' ? data[datum - 1]?.link ?? '' : '';
+};
+
+export type ComplianceData = {
+    name: string;
+    passing: number;
+    link: string;
+}[];
+
+const mockComplianceData = [
+    {
+        name: 'staging',
+        passing: 100,
+        link: '',
+    },
+    {
+        name: 'production',
+        passing: 80,
+        link: '',
+    },
+    {
+        name: 'payments',
+        passing: 73,
+        link: '',
+    },
+    {
+        name: 'patient-charts',
+        passing: 69,
+        link: '',
+    },
+    {
+        name: 'another-cluster',
+        passing: 67,
+        link: '',
+    },
+    {
+        name: 'cluster-name',
+        passing: 39,
+        link: '',
+    },
+];
+
+function ComplianceByCluster() {
+    const [complianceData] = useState(mockComplianceData);
+    const [widgetContainer, setWidgetContainer] = useState<HTMLDivElement | null>(null);
+    const widgetContainerResizeEntry = useResizeObserver(widgetContainer);
+
+    function getBarColor(percent) {
```

Comment:
Style question for your consideration about pro and con to move pure function from component function scope to file scope?

Also, classic compliance has a different mapping. It is a worthwhile question for PM and UX about pro and con to align 1.0 with 2.0 mapping similar to how we have aligned colors with Workflow CVEs.
https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/Compliance/widgets/colorsForCompliance.ts#L9-L23

Reply:
Good callout, I didn't even know that existed there. I'd hate to copy that file over and have a duplicate but I don't want to be importing anything from 1.0. Any suggestions?